insalan/tournament/models.py

```diff
@@ -571,6 +643,35 @@ def as_user(self) -> User:
     def get_team(self):
         """Return the Team object of the current team"""
         return self.team
+    
+    def get_pseudo(self) -> str:
+        """Return the pseudo of the player"""
+        return self.pseudo
 
+    def clean(self):
+        """
+        Assert that the user associated with the provided manager does not already
+        exist in any team of any tournament of the event
+        """
```

Comment:
no, as far as I understand the django documentation, this function is part of the validation process of objects in modelForms. By the way I struggle understanding the validation process because in the DRF documentation it say that the validation only take place in the serializer so this clean function will not be call but since we use the admin panel it seems that it will call this function when we create objects from the admin panel. So the validation can occur in two different places.

Reply:
Alright, it once again is a dirty way of django of doing it

Reply:
Django is an amazing(ly comfusing) framework, as usual (but he's right)